wgsl/index.bs

```diff
@@ -435,61 +435,111 @@ A <dfn>token</dfn> is a contiguous sequence of characters forming one of:
-
-The form of a [=numeric literal=] is defined via pattern-matching:
+* A <dfn noexport>boolean literal</dfn> is either `true` or `false`.
+* A <dfn>numeric literal</dfn> is either an [=integer literal=] or a [=floating point literal=],
+    and is used to represent a number.
 
 <div class='syntax' noexport='true'>
-  <dfn for=syntax>decimal_float_literal</dfn> :
+  <dfn for=syntax>bool_literal</dfn> :
 
-    | `/((-?[0-9]*\.[0-9]+|-?[0-9]+\.[0-9]*)((e|E)(\+|-)?[0-9]+)?f?)|(-?[0-9]+(e|E)(\+|-)?[0-9]+f?)/`
-</div>
-<div class='syntax' noexport='true'>
-  <dfn for=syntax>hex_float_literal</dfn> :
+    | [=syntax/true=]
 
-    | `/-?0[xX]((([0-9a-fA-F]*\.[0-9a-fA-F]+|[0-9a-fA-F]+\.[0-9a-fA-F]*)((p|P)(\+|-)?[0-9]+f?)?)|([0-9a-fA-F]+(p|P)(\+|-)?[0-9]+f?))/`
+    | [=syntax/false=]
 </div>
+
+The form of a [=numeric literal=] is defined via pattern-matching.
+
+An <dfn>integer literal</dfn> is:
+* An integer specified as any of:
+    * `0`
+    * A sequence of decimal digits, where the first digit is not `0`.
+    * `0x` or `0X` followed by a sequence of hexadecimal digits.
+* Then an optional `i` or `u` suffix.
+
 <div class='syntax' noexport='true'>
   <dfn for=syntax>int_literal</dfn> :
 
-    | `/-?0[xX][0-9a-fA-F]+|0|-?[1-9][0-9]*/`
+    | `/(0[xX][0-9a-fA-F]+|0|[1-9][0-9]*)[iu]?/`
+</div>
+
+An <dfn>floating point literal</dfn> is either a decimal floating point literal
+or a hexadecimal floating point literal.
+* A <dfn noexport>decimal floating point literal</dfn> is:
+    * A mantissa, specified as a sequence of digits, with an optional decimal point (`.`) somewhere among them.
+    * Then an optional exponent suffix consisting of:
+        * `e` or `E`.
+        * Then an exponent specified as an decimal number with an optional leading sign (`+` or `-`).
+    * Then an optional `f` suffix.
+    * At least one of the decimal point, or the exponent, or the `f` suffix must be present.
+         If none are, then the token is instead an [=integer literal=].
+    * The value of the literal is the value of the mantissa multiplied by 10 to the power of the exponent.
+         When no exponent is specified, an exponent of 0 is assumed.
+
+* A <dfn noexport>hexadecimal floating point literal</dfn> is:
+    * A `0x` or `0X` prefix
+    * Then a mantissa, specified as a sequence of hexadecimal digits, with an optional hexadecimal point (`.`) somewhere among them.
+    * Then an optional exponent suffix consisting of:
+        * `p` or `P`
+        * Then an exponent specified as an decimal number with an optional leading sign (`+` or `-`).
+    * Then an optional `f` suffix.
+    * At least one of the hexadecimal point, or the exponent, or the `f` suffix must be present.
+         If none are, then the token is instead an [=integer literal=].
+    * The value of the literal is the value of the mantissa multiplied by 2 to the power of the exponent.
+         When no exponent is specified, an exponent of 0 is assumed.
```

Comment:
This prose doesn't match the regexp. The prose permits `0x1f`, but the regular expression does not.

Illustrated: https://www.debuggex.com/r/AlaAnntP9cuSM4lB

If you compare the decimal and the hex regexps in two tabs using that site, it's very clear where they differ.

Reply:
I was naively assuming the regexp had the bug, but since an `f` suffix is ambiguous with a hex digit unless there's an exponent or a decimal point, the regexp looks right, and it seems the prose is what needs to change.